<a href="https://colab.research.google.com/github/sebekpro/python-krakowiak/blob/main/250-data-science-bootcamp-w-jezyku-python/181_190_exercises.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Pandas

### Spis treści:
* [Import biblioteki](#0)
* [Ćwiczenie 181](#1)
* [Ćwiczenie 182](#2)
* [Ćwiczenie 183](#3)
* [Ćwiczenie 184](#4)
* [Ćwiczenie 185](#5)
* [Ćwiczenie 186](#6)
* [Ćwiczenie 187](#7)
* [Ćwiczenie 188](#8)
* [Ćwiczenie 189](#9)
* [Ćwiczenie 190](#10)

### <a name='0'></a>Import biblioteki

In [ ]:
import numpy as np
import pandas as pd
import plotly.express as px

np.__version__

'1.26.4'

Uruchom poniższą instrukcję aby pobrać plik _insurance.csv_ do katalogu roboczego.

In [ ]:
!wget -q https://storage.googleapis.com/esmartdata-courses-files/ml-course/insurance.csv

### <a name='1'></a> Ćwiczenie 181
Wczytaj plik _insurance.csv_ do obiektu DataFrame o nazwie _df_raw_ i wyświetl kilka początkowych wierszy.

In [ ]:
# tutaj wpisz rozwiązanie
df_raw = pd.read_csv('insurance.csv')
df_raw.head()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

Skopiuj obiekt _df_raw_ do zmiennej _df_ i wyświetl podstawowe informacje o obiekcie _df_.

In [ ]:
# tutaj wpisz rozwiązanie
df = df_raw.copy()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


### <a name='2'></a> Ćwiczenie 182
Sprawdź czy istnieją zduplikowane wiersze w obiekcie _df_. Jeśli tak, wyświetl te wiersze.

In [ ]:
# tutaj wpisz rozwiązanie
print(df.duplicated().sum())
print(df[df.duplicated()])


1
     age   sex    bmi  children smoker     region    charges
581   19  male  30.59         0     no  northwest  1639.5631


Usuń duplikaty w obiekcie _df_.

In [ ]:
# tutaj wpisz rozwiązanie
df.drop_duplicates(inplace = True)
print(df.duplicated().sum())
print(df[df.duplicated()])

0
Empty DataFrame
Columns: [age, sex, bmi, children, smoker, region, charges]
Index: []


### <a name='3'></a> Ćwiczenie 183
Wyświetl z obiektu _df_ zmienne z typem _object_.

__Wskazówka:__ Użyj metody _pd.DataFrame.select_dtypes()_

In [ ]:
# tutaj wpisz rozwiązanie
df.select_dtypes(include = 'object')

sex smoker     region
0     female    yes  southwest
1       male     no  southeast
2       male     no  southeast
3       male     no  northwest
4       male     no  northwest
...      ...    ...        ...
1333    male     no  northwest
1334  female     no  northeast
1335  female     no  southeast
1336  female     no  southwest
1337  female    yes  northwest

[1337 rows x 3 columns]

Przekształć te zmienne do zmiennych kategorycznych.

__Wskazówka:__ Użyj metody _pd.DataFrame.astype()_.

In [ ]:
# tutaj wpisz rozwiązanie
df['sex'] = df['sex'].astype('category')
df['smoker'] = df['smoker'].astype('category')
df['region'] = df['region'].astype('category')
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1337 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   age       1337 non-null   int64   
 1   sex       1337 non-null   category
 2   bmi       1337 non-null   float64 
 3   children  1337 non-null   int64   
 4   smoker    1337 non-null   category
 5   region    1337 non-null   category
 6   charges   1337 non-null   float64 
dtypes: category(3), float64(2), int64(2)
memory usage: 56.6 KB


Wyświetl z obiektu _df_ zmienne numeryczne (_float_, _int_).

__Wskazówka:__ Użyj metody _pd.DataFrame.select_dtypes()_

In [ ]:
# tutaj wpisz rozwiązanie
df.select_dtypes(include = 'number')

age     bmi  children      charges
0      19  27.900         0  16884.92400
1      18  33.770         1   1725.55230
2      28  33.000         3   4449.46200
3      33  22.705         0  21984.47061
4      32  28.880         0   3866.85520
...   ...     ...       ...          ...
1333   50  30.970         3  10600.54830
1334   18  31.920         0   2205.98080
1335   18  36.850         0   1629.83350
1336   21  25.800         0   2007.94500
1337   61  29.070         0  29141.36030

[1337 rows x 4 columns]

Znajdź nazwy zmiennych numerycznych (_float_, _int_) z obiektu _df_.

In [ ]:
# tutaj wpisz rozwiązanie
df.select_dtypes(include = 'number').columns

Index(['age', 'bmi', 'children', 'charges'], dtype='object')

### <a name='4'></a> Ćwiczenie 184

Wykonaj poniższe kroki:

* przypisz do zmiennej _df_cat_ zmienne kategoryczne z obiektu _df_ i utwórz ich kopię.

* przypisz do zmiennej _df_num_ zmienne numeryczne z obiektu _df_ i utwórz ich kopię.

In [ ]:
# tutaj wpisz rozwiązanie
df_cat = df.select_dtypes(include = 'category').copy()
df_num = df.select_dtypes(include = 'number').copy()

Wyświetl 5 pierwszych wierszy obiektu _df_cat_.

In [ ]:
# tutaj wpisz rozwiązanie
df_cat.head()

sex smoker     region
0  female    yes  southwest
1    male     no  southeast
2    male     no  southeast
3    male     no  northwest
4    male     no  northwest

Wyświetl 5 pierwszych wierszy obiektu _df_num_.

In [ ]:
# tutaj wpisz rozwiązanie
df_num.head()

age     bmi  children      charges
0   19  27.900         0  16884.92400
1   18  33.770         1   1725.55230
2   28  33.000         3   4449.46200
3   33  22.705         0  21984.47061
4   32  28.880         0   3866.85520

### <a name='5'></a> Ćwiczenie 185
Wyświetl podstawowe statystyki obiektu _df_cat_.

In [ ]:
# tutaj wpisz rozwiązanie
df_cat.describe()

sex smoker     region
count   1337   1337       1337
unique     2      2          4
top     male     no  southeast
freq     675   1063        364

Wyświetl podstawowe statystyki obiektu _df_num_.

In [ ]:
# tutaj wpisz rozwiązanie
df_num.describe()

age          bmi     children       charges
count  1337.000000  1337.000000  1337.000000   1337.000000
mean     39.222139    30.663452     1.095737  13279.121487
std      14.044333     6.100468     1.205571  12110.359656
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.290000     0.000000   4746.344000
50%      39.000000    30.400000     1.000000   9386.161300
75%      51.000000    34.700000     2.000000  16657.717450
max      64.000000    53.130000     5.000000  63770.428010

### <a name='6'></a> Ćwiczenie 186
Wyświetl podstawowe statystyki obiektu _df_ dla zmiennych numerycznych i dokonaj transponowania otrzymanego obiektu _DataFrame_.

In [ ]:
# tutaj wpisz rozwiązanie
df.describe().T

count          mean           std        min       25%        50%  \
age       1337.0     39.222139     14.044333    18.0000    27.000    39.0000   
bmi       1337.0     30.663452      6.100468    15.9600    26.290    30.4000   
children  1337.0      1.095737      1.205571     0.0000     0.000     1.0000   
charges   1337.0  13279.121487  12110.359656  1121.8739  4746.344  9386.1613   

                  75%          max  
age          51.00000     64.00000  
bmi          34.70000     53.13000  
children      2.00000      5.00000  
charges   16657.71745  63770.42801

Wyświetl podstawowe statystyki obiektu _df_ dla zmiennych numerycznych, dokonaj transponowania otrzymanego obiektu _DataFrame_ i wytnij dwie columny _mean_ i _std_.

In [ ]:
# tutaj wpisz rozwiązanie
stats = df.describe().T[["mean","std"]]
stats

mean           std
age          39.222139     14.044333
bmi          30.663452      6.100468
children      1.095737      1.205571
charges   13279.121487  12110.359656

Wyświetl podstawowe statystyki obiektu _df_ dla zmiennych kategorycznych i dokonaj transponowania otrzymanego obiektu _DataFrame_.

In [ ]:
# tutaj wpisz rozwiązanie
df.describe(include = 'category').T

count unique        top  freq
sex     1337      2       male   675
smoker  1337      2         no  1063
region  1337      4  southeast   364

Sprawdź, czy są jakieś braki danych w obiekcie _df_.

In [ ]:
# tutaj wpisz rozwiązanie
df.isna().sum()

age         0
sex         0
bmi         0
children    0
smoker      0
region      0
charges     0
dtype: int64

### <a name='6'></a> Ćwiczenie 186
Wykorzystując bibliotekę _plotly express_ zbuduj histogram zmiennej _charges_.

Użyj funkcji _px.histogram()_ z następującymi parametrami:
* width=800
* height=400
* nbins=50
* facet_col='smoker'
* facet_row='sex'

In [ ]:
# tutaj wpisz rozwiązanie
px.histogram(df, x = 'charges',
             width = 800,           # Szerokość wykresu ustawiona na 800 pikseli
             height = 400,          # Wysokość wykresu ustawiona na 400 pikseli
             nbins = 50,            # Liczba przedziałów (koszyków) histogramu wynosi 50
             facet_col = 'smoker',  # Histogram podzielony na kolumny według zmiennej 'smoker'
             facet_row = 'sex',     # Histogram podzielony na wiersze według zmiennej 'sex'
             title = 'Histogram zmiennej charges')

* Parametry width i height: Ustawiają rozmiary wykresu na 800 pikseli szerokości i 400 pikseli wysokości.
* Parametr nbins: Definiuje liczbę przedziałów (koszyków) histogramu, co wpływa na szczegółowość rozkładu.
* Parametry facet_col oraz facet_row: Umożliwiają podział wykresu na mniejsze wykresy (facety) według zmiennych smoker (w kolumnach) oraz sex (w wierszach). Dzięki temu możemy zobaczyć rozkład zmiennej charges w podziale na kategorie palenia i płci.
* Tytuł wykresu: Parametr title ustawia tytuł wykresu na "Histogram zmiennej charges".

Wykorzystując bibliotekę _plotly express_ znajdź rozkład zmiennej _smoker_.

Użyj funkcji _px.histogram()_ z następującymi parametrami:
* width=800
* height=400
* facet_col='sex'
* color='sex'

In [ ]:
# tutaj wpisz rozwiązanie
fig = px.histogram(df, x = 'smoker',
                   width = 800,             # Szerokość wykresu ustawiona na 800 pikseli
                   height = 400,            # Wysokość wykresu ustawiona na 400 pikseli
                   facet_col = 'sex',       # Podział wykresu na kolumny według zmiennej 'sex'
                   color = 'sex',           # Kolorowanie danych według zmiennej 'sex'
                   title = 'Histogram zmiennej smoker'
                   )
fig.show()

* Funkcja px.histogram(): Generuje histogram dla zmiennej smoker z obiektu DataFrame df.
* Parametr x='smoker': Określa, że oś X wykresu reprezentuje zmienną smoker.
* Parametry width i height: Ustawiają rozmiar wykresu na 800 pikseli szerokości i 400 pikseli wysokości.
* Parametr facet_col='sex': Dzieli wykres na osobne kolumny w zależności od wartości zmiennej sex.
* Parametr color='sex': Wykorzystuje zmienną sex do kolorowania wykresu, co umożliwia lepsze porównanie rozkładu w poszczególnych grupach.
* Tytuł wykresu: Ustawia tytuł na "Rozkład zmiennej smoker".
* fig.show(): Wyświetla interaktywny wykres w środowisku obsługującym Plotly.

### <a name='7'></a> Ćwiczenie 187
Skonwertuj zmienne kategoryczne obietku _df_ do kodowania 0-1 i przypisz do zmiennej _df_dummies_.

__Wskazówka:__ Użyj funkcji _pd.get_dummies()_ z parametrem _drop_first=True_.

In [ ]:
# tutaj wpisz rozwiązanie
df_dummies = pd.get_dummies(df,drop_first = True)
print(df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 1337 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   age       1337 non-null   int64   
 1   sex       1337 non-null   category
 2   bmi       1337 non-null   float64 
 3   children  1337 non-null   int64   
 4   smoker    1337 non-null   category
 5   region    1337 non-null   category
 6   charges   1337 non-null   float64 
dtypes: category(3), float64(2), int64(2)
memory usage: 56.6 KB
None


* pd.get_dummies(df, drop_first=True): Funkcja ta konwertuje wszystkie zmienne kategoryczne w obiekcie df na kolumny binarne (0 lub 1).
* drop_first=True: Ustawienie tego parametru powoduje, że z każdej zmiennej kategorycznej pomijana jest pierwsza kategoria, co redukuje nadmiarowość danych.
* Wynikowa DataFrame jest przypisana do zmiennej df_dummies.

Funkcja **pd.get_dummies()** przekształca kolumny kategoryczne w DataFrame na zestaw kolumn zero-jedynkowych (dummy variables), gdzie każda nowa kolumna reprezentuje jedną z kategorii danej zmiennej. Używając parametru **drop_first=True**, dla każdej zmiennej kategorycznej zostaje pominięta pierwsza kategoria, aby uniknąć redundancji (tzw. dummy variable trap).

W przypadku:

* **sex**: Oryginalna zmienna miała dwie kategorie (np. "female" i "male"). Po kodowaniu, jedna z nich (np. "female") jest pominięta, a w rezultacie powstaje kolumna **sex_male**. Wartość **True** w tej kolumnie oznacza, że dany wiersz dotyczy osoby płci męskiej.
* **smoker**: Podobnie, dla zmiennej **smoker** z kategoriami (np. "no" i "yes") zostaje utworzona tylko jedna kolumna, czyli **smoker_yes**. Wartość **True** wskazuje, że osoba pali.
* **region**: Jeśli zmienna **region** miała cztery kategorie (np. "northeast", "northwest", "southeast", "southwest"), to przy pominięciu pierwszej kategorii (najprawdopodobniej "northeast") powstaną trzy nowe kolumny: region_northwest, **region_southeast** i **region_southwest**. Każda z nich przyjmuje wartość **True** dla wierszy, które odpowiadają danej kategorii regionu.

W rezultacie, poza oryginalnymi zmiennymi numerycznymi, otrzymujesz 5 dodatkowych kolumn reprezentujących zakodowane zmienne kategoryczne.

### <a name='8'></a> Ćwiczenie 188
Wyznacz korelacje zmiennych w obiekcie _df_dummies_ i przypisz do zmiennej _corr_.

In [ ]:
# tutaj wpisz rozwiązanie
corr = df_dummies.corr()
corr

age       bmi  children   charges  sex_male  \
age               1.000000  0.109344  0.041536  0.298308 -0.019814   
bmi               0.109344  1.000000  0.012755  0.198401  0.046397   
children          0.041536  0.012755  1.000000  0.067389  0.017848   
charges           0.298308  0.198401  0.067389  1.000000  0.058044   
sex_male         -0.019814  0.046397  0.017848  0.058044  1.000000   
smoker_yes       -0.025587  0.003746  0.007331  0.787234  0.076596   
region_northwest  0.001495 -0.136138  0.026044 -0.038695 -0.012482   
region_southeast -0.012311  0.270057 -0.023492  0.073578  0.017578   
region_southwest  0.009415 -0.006211  0.021538 -0.043637 -0.003767   

                  smoker_yes  region_northwest  region_southeast  \
age                -0.025587          0.001495         -0.012311   
bmi                 0.003746         -0.136138          0.270057   
children            0.007331          0.026044         -0.023492   
charges             0.787234         -0.038695          0.073578   
sex_male            0.076596         -0.012482          0.017578   
smoker_yes          1.000000         -0.036321          0.068282   
region_northwest   -0.036321          1.000000         -0.345909   
region_southeast    0.068282         -0.345909          1.000000   
region_southwest   -0.037168         -0.320493         -0.346614   

                  region_southwest  
age                       0.009415  
bmi                      -0.006211  
children                  0.021538  
charges                  -0.043637  
sex_male                 -0.003767  
smoker_yes               -0.037168  
region_northwest         -0.320493  
region_southeast         -0.346614  
region_southwest          1.000000

* df_dummies.corr(): Ta metoda oblicza macierz korelacji Pearsona dla wszystkich kolumn w DataFrame df_dummies.
* Wynikowa macierz, która zawiera współczynniki korelacji między zmiennymi, jest przypisywana do zmiennej corr.

Wytnij kolumnę z obiektu _corr_ dla zmiennej _charges_ (jako obiekt DataFrame) i posortuj wartości malejąco.

In [ ]:
# tutaj wpisz rozwiązanie
corr_charges = corr[['charges']].sort_values(by = 'charges', ascending = False)
corr_charges

charges
charges           1.000000
smoker_yes        0.787234
age               0.298308
bmi               0.198401
region_southeast  0.073578
children          0.067389
sex_male          0.058044
region_northwest -0.038695
region_southwest -0.043637

* corr[['charges']] – Wycinamy kolumnę charges z obiektu corr. Użycie podwójnych nawiasów zwraca wynik jako DataFrame.
* .sort_values(by='charges', ascending=False) – Sortujemy wartości w kolumnie charges w porządku malejącym.


Wykorzystując bibliotekę _plotly express_ utwórz wykres słupkowy dla tak posortowanych wartości. Pomiń pierwszy element obiektu.

In [ ]:
# tutaj wpisz rozwiązanie
filtered_corr = corr_charges.iloc[1:]
fig = px.bar(filtered_corr, x = filtered_corr.index, y = 'charges',
             title = 'Korelacje zmiennych z charges (pomijając pierwszy element)',
             labels = {'index': 'Zmienne', 'charges': 'Współczynnik korelacji'})
fig.show()

* filtered_corr = corr_charges.iloc[1:]: Ta linia usuwa pierwszy wiersz z DataFrame, który odpowiada korelacji zmiennej charges z samą sobą.
* px.bar(...): Funkcja tworzy wykres słupkowy, gdzie:
x=filtered_corr.index ustawia etykiety słupków (nazwy zmiennych).
y='charges' określa wartości na osi Y (współczynniki korelacji z charges).
* fig.show(): Wyświetla interaktywny wykres.

### <a name='9'></a> Ćwiczenie 189
Skopiuj obiekt _df_dummies_ do zmiennej _data_. Następnie wyrwij z tego obiektu kolumnę _charges_ i przypisz do zmiennej _target_.

In [ ]:
# tutaj wpisz rozwiązanie
data = df_dummies.copy()
target = data.pop('charges')

* data = df_dummies.copy() – Tworzy głęboką kopię obiektu df_dummies i przypisuje ją do zmiennej data, co umożliwia dalsze operacje bez modyfikowania oryginału.
* target = data.pop('charges') – Metoda pop() wycina kolumnę charges z DataFrame data i przypisuje ją do zmiennej target. Dzięki temu kolumna charges zostaje usunięta z obiektu data.

Wyświetl 5 pierwszych wierszy obiektu _data_. Upewnij się, że nie ma w nim kolumny _charges_.

In [ ]:
# tutaj wpisz rozwiązanie
data.head()

age     bmi  children  sex_male  smoker_yes  region_northwest  \
0   19  27.900         0     False        True             False   
1   18  33.770         1      True       False             False   
2   28  33.000         3      True       False             False   
3   33  22.705         0      True       False              True   
4   32  28.880         0      True       False              True   

   region_southeast  region_southwest  
0             False              True  
1              True             False  
2              True             False  
3             False             False  
4             False             False

Wyświetl 5 pierwszych wierszy obiektu _target_.

In [ ]:
# tutaj wpisz rozwiązanie
target.head()

0    16884.92400
1     1725.55230
2     4449.46200
3    21984.47061
4     3866.85520
Name: charges, dtype: float64

### <a name='10'></a> Ćwiczenie 190
Wykorzystując bibliotekę _scikit-learn_ oraz funkcję _train_test_split_ podziel dane z obiektów _data_ i _target_ na zbiór treningowy (_X\_train_, _y\_train_) oraz zbiór testowy (_X\_test_, _y\_test_). Do zbioru testowego przydziel 20% próbek.

Wyświetl rozmiar obiektów:
* _X\_train_
* _X\_test_
* _y\_train_
* _y\_test_

In [ ]:
# tutaj wpisz rozwiązanie
from sklearn.model_selection import train_test_split

# Podział danych na zbiór treningowy i testowy (20% próbek w zbiorze testowym)
X_train, X_test, y_train, y_test = train_test_split(data, target, test_size = 0.20, random_state = 42)

# Wyświetlenie rozmiaru obiektów
print("Rozmiar X_train", X_train.shape)
print("Rozmiar X_test", X_test.shape)
print("Rozmiar y_train", y_train.shape)
print("Rozmiar y_test", y_test.shape)


Rozmiar X_train (1069, 8)
Rozmiar X_test (268, 8)
Rozmiar y_train (1069,)
Rozmiar y_test (268,)


* Importujemy funkcję train_test_split z modułu sklearn.model_selection.
* Funkcja train_test_split dzieli dane na zbiory treningowy i testowy. Parametr test_size=0.20 oznacza, że 20% danych trafia do zbioru testowego.
* random_state=42 ustawia ziarno generatora liczb losowych, aby wynik podziału był powtarzalny.
* Po podziale wyświetlamy kształty (rozmiary) obiektów: X_train, X_test, y_train oraz y_test.

Wybierając wartość **42**, gwarantujemy, że podział danych na zbiory treningowy i testowy będzie taki sam przy każdym wykonaniu, co ułatwia porównywanie eksperymentów i debugowanie. Sam numer **42** nie ma specjalnego znaczenia technicznego – jest często wybierany ze względu na kultowy odniesienie do "odpowiedzi na pytanie o życie, wszechświat i całą resztę" z książki Douglasa Adamsa, ale pod kątem funkcjonalnym mógłby to być dowolny inny numer całkowity.

Wynikowe wymiary oznaczają, że:

* X_train (1069, 8): Zbiór treningowy zawiera 1069 próbek (wierszy) oraz 8 cech (kolumn). Każda próbka reprezentuje obserwację (np. osobę, przypadek) opisaną 8 cechami.
* X_test (268, 8): Zbiór testowy zawiera 268 próbek z tymi samymi 8 cechami. Te dane są używane do oceny modelu po jego wytrenowaniu.
* y_train (1069,): Zbiór etykiet (target) dla danych treningowych składa się z 1069 wartości, czyli jednej wartości dla każdej próbki w zbiorze treningowym.
* y_test (268,): Zbiór etykiet dla danych testowych składa się z 268 wartości.

Oznacza to, że dane zostały podzielone na 80% próbek (1069) przeznaczonych do trenowania modelu oraz 20% próbek (268) do jego testowania, przy czym każdy przykład w zbiorze cech odpowiada jednej wartości docelowej (target).

Zapisz obiekty:
* _X\_train_
* _X\_test_
* _y\_train_
* _y\_test_  

do plików csv odpowiednio o nazwie:
* X_train.csv
* X_test.csv
* y_train.csv
* y_test.csv


In [ ]:
# tutaj wpisz rozwiązanie
X_train.to_csv('X_train.csv', index = False)
X_test.to_csv('X_test.csv', index = False)
y_train.to_csv('y_train.csv', index = False)
y_test.to_csv('y_test.csv', index = False)

Uruchom poniższą komórkę w celu sprawdzenia.

In [ ]:
!head -5 X_test.csv
print('-----------------------\n')
!head -5 X_train.csv
print('-----------------------\n')
!head -5 y_train.csv
print('-----------------------\n')
!head -5 y_test.csv

age,bmi,children,sex_male,smoker_yes,region_northwest,region_southeast,region_southwest
49,22.515,0,True,False,False,False,False
29,25.6,4,False,False,False,False,True
51,36.385,3,False,False,True,False,False
31,34.39,3,True,True,True,False,False
-----------------------

age,bmi,children,sex_male,smoker_yes,region_northwest,region_southeast,region_southwest
23,24.51,0,True,False,False,False,False
21,25.745,2,True,False,False,False,False
52,37.525,2,False,False,True,False,False
63,41.47,0,True,False,False,True,False
-----------------------

charges
2396.0959
3279.86855
33471.97189
13405.3903
-----------------------

charges
8688.85885
5708.867
11436.73815
38746.3551
